app/controllers/qsets_controller.rb

```diff
@@ -12,31 +12,33 @@ def index
 
   # handles the request to show all questions in a qset
   def show
-    @feedback_active = !!current_user
-    # sorts by default by net votes; secondary sort by create date
-    @questions = Question.includes(:answers).where(qset_id: @qset.id).plusminus_tally.order(created_at: :desc)
-    if can? :see_all_questions, @qset
-      @filter_mine = true if cookies[:all_mine_other_filter] == 'mine'
-      @filter_other = true if cookies[:all_mine_other_filter] == 'other'
-      @filter_all = true unless @filter_mine or @filter_other
-    else
-      # show only the current user's questions
-      cookies[:all_mine_other_filter] = 'mine'
-      @filter_mine = true
-      @questions = @questions.where(user_id: current_user)
-    end
     @qsets = @qset.children
-    # a hash of qset question counts keyed by qset id
-    @subset_question_counts = @qsets.map do |s|
-      count = 0
-      s.self_and_descendants.each do |q|
-        scope = q.questions
-        scope = scope.where(user_id: current_user) if cannot? :see_all_questions, q
-        count += scope.count
+      # a hash of qset question counts keyed by qset id
+      @subset_question_counts = @qsets.map do |s|
+        count = 0
+        s.self_and_descendants.each do |q|
+          scope = q.questions
+          scope = scope.where(user_id: current_user) if cannot? :see_all_questions, q
+          count += scope.count
+        end
+        [s.id, count]
+      end.to_h
+    # by default show question page
+    if @qset.settings(:permissions).qset_type != 'subset' 
```

Comment:
can this be `subsets` (plural), to match `questions`? This can all change later, but might as well get it right ahead of time. Use your judgment but keep in mind that:
- this is a property of a qset
- it dictates what users will _see_ when they access the qset
- the possible 'values' will be seeing a list of questions, a list of subsets, or a mixed view with both

... choose your property name and the possible string values carefully with that in mind, so it's as clear/self-documenting as possible. It might be nice, since only one string value is called out here (`'subset(s)'`), to have a comment somewhere in the code, maybe in application.rb (where the default is defined) list the possible values, just for reference.